### ISSUE TITLE
Segmentation Prediction: Saved Text File Normalisation

### ISSUE THREAD

**Issue by Bampson** [ultralytics/yolov5#10196]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Maybe a bug? Noticing that the txt file generated after segmentation prediction is outputting different values to the object detection detection:

eg.
`3 780 588 778 590 772 590 772 593 768 598 765 598 762 600 762 603 740 ... 0.464866`

Should these be normalised under 1?

### Additional

using: `--save-conf --save-txt`

**Comment by glenn-jocher** [comment]
@Bampson yes these should be normalized. If they are not then there is a bug.

**Comment by glenn-jocher** [comment]
@Bampson can reproduce bug. Have added bug and TODO labels to this.

<img width="1721" alt="Screenshot 2022-11-18 at 13 43 36" src="https://user-images.githubusercontent.com/26833433/202707770-0c8fd23d-cbb0-43e1-a4e7-6fb456ad5dd6.png">

**Comment by glenn-jocher** [comment]
@Bampson good news 😃! Your original issue may now be fixed ✅ in PR #10205. Segmentation predictions are now outputted in normalized coordinates just like detection predictions.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by Bampson** [comment]
@glenn-jocher Amazing work thanks for that. This segmentation you have implemented is amazing stuff.

@WongVi Yeah a custom one, my dataset looks something like this when its ready:
```
dataset
├── images
│   ├── train
│   │   └── IMG_123.jpg
│   └── val
│       └── IMG_456.jpg
└── labels
    ├── train
    │   └── IMG_123.txt
    └── val
        └── IMG_456.txt
```

You can use https://github.com/ultralytics/JSON2YOLO to convert datasets from a bunch of different JSON formats such as COCO.
Just created a new yaml based on the COCO128-seg.yaml to fit my classes and dataset path.

**Comment by ryouchinsa** [comment]
@WongVi, @Bampson 

You can use Labelme to label polygons and convert the JSON files to the YOLO format using JSON2YOLO.
https://github.com/wkentaro/labelme
https://github.com/ultralytics/JSON2YOLO

Although this is not an open source program, with RectLabel, you can label polygons and keypoints in the YOLO format.
https://rectlabel.com/help#xml_to_yolo

Please give us your feedback about the combination of YOLOv5 and RectLabel.

**Comment by glenn-jocher** [comment]
@Bampson thank you for your kind words, but the credit truly goes to the incredible work of the YOLO community and the diligent efforts of the Ultralytics team in advancing the YOLOv5 project. We greatly appreciate your contributions and feedback, and we're delighted to hear that the segmentation functionality has been beneficial to you. Should you encounter any further questions or require assistance, please don't hesitate to reach out. Happy coding and best of luck with your custom dataset!
